src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -601,13 +594,10 @@ private static bool StringRequiresHtmlDecoding(string s)
 
         #endregion
 
-        #region UrlDecoder nested class
-
-        // *** Source: alm/tfs_core/Framework/Common/UriUtility/HttpUtility.cs
-        // This specific code was copied from above ASP.NET codebase.
+        #region UrlDecoder nested struct
 
-        // Internal class to facilitate URL decoding -- keeps char buffer and byte buffer, allows appending of either chars or bytes
-        private class UrlDecoder
+        // Internal struct to facilitate URL decoding -- keeps char buffer and byte buffer, allows appending of either chars or bytes
+        private struct UrlDecoder
```

Comment:
Whoops, looks like I've gotten a little too used to using GitHub Edtior :smile: I'll compile and test this locally before submitting any more changes, sorry.